### PR TITLE
Fixed #29249 -- Added --allow_unicode option to dumpdata to allow unicode JSON or YAML.

### PR DESCRIPTION
**Pull request by hramezani**
[Ticket 29249](https://code.djangoproject.com/ticket/29249).
Continue #9818. I added documentation.

According to [this comment](https://code.djangoproject.com/ticket/29249#comment:3), please suggest the new option name I'll change it.

### CODE REVIEW

File: django/core/management/commands/dumpdata.py [hunk 1]
```diff
@@ -63,6 +63,10 @@ def add_arguments(self, parser):
             '-o', '--output',
             help='Specifies file to which the output is written.'
         )
+        parser.add_argument(
+            '--allow-unicode', action='store_false', dest='allow_unicode',
```
Comment (charettes) [inline django/core/management/commands/dumpdata.py:67]: I think it should be `store_true` and `dest` is redundant.

```suggestion
            '--allow-unicode', action='store_true',
```

Wonder if a command test should be added to test that?

Comment (hramezani) [reply django/core/management/commands/dumpdata.py]: Thanks @charettes,
I've addressed your suggestion and added requested tests as well.

File: docs/ref/django-admin.txt [hunk 2]
```diff
@@ -339,6 +339,12 @@ standard output.
 When this option is set and ``--verbosity`` is greater than 0 (the default), a
 progress bar is shown in the terminal.
 
+.. django-admin-option:: --allow-unicode
+
+.. versionadded:: 3.1
+
+Allows for unicode characters in output.
```
Comment (jdufresne) [inline docs/ref/django-admin.txt:346]: Unicode is a proper noun. Same comment for the release notes.

```suggestion
Allows for Unicode characters in output.
```

File: tests/serializers/test_json.py [hunk 3]
```diff
@@ -319,3 +335,23 @@ def test_timedelta(self):
...
+        for allow_unicode, expected in tests:
+            with self.subTest(allow_unicode=allow_unicode, expected=expected):
+                out = StringIO()
+                call_command(
+                    'dumpdata',
+                    'serializers.actor',
+                    allow_unicode=allow_unicode,
+                    stdout=out,
+                )
+                self.assertTrue(expected in out.getvalue().strip())
```
Comment (jdufresne) [inline tests/serializers/test_json.py:357]: Can use `asserIn` here for a more specific assertion method. Same for other tests.

```suggestion
                self.assertIn(expected, out.getvalue().strip())
```

**Comment by hramezani** [comment]
Thanks @jdufresne for review. I've addressed them.

**Comment by felixxm** [comment]
@hramezani Thanks for this patch :+1: , however I believe that we should solve by making serializers behavior consistent.

**Comment by felixxm** [comment]
Closing per [comment](https://code.djangoproject.com/ticket/29249#comment:6).

**Pull request closed without merging**